schema.js

```diff
@@ -1,102 +1,337 @@
+    "ais.currentRefinements": {
+        "value": 7,
+        "options": {}
+    },
+    "ais.geoSearch": {
+        "value": 8,
+        "options": {}
+    },
+    "ais.hierarchicalMenu": {
+        "value": 9,
+        "options": {}
+    },
+    "ais.hits": {
+        "value": 10,
+        "options": {}
+    },
+    "ais.hitsPerPage": {
+        "value": 11,
+        "options": {}
+    },
+    "ais.index": {
+        "value": 12,
+        "options": {}
+    },
+    "ais.infiniteHits": {
+        "value": 13,
+        "options": {}
+    },
+    "ais.menu": {
+        "value": 14,
+        "options": {}
+    },
+    "ais.numericMenu": {
+        "value": 15,
+        "options": {}
+    },
+    "ais.pagination": {
+        "value": 16,
+        "options": {}
+    },
+    "ais.places": {
+        "value": 17,
+        "options": {}
+    },
+    "ais.poweredBy": {
+        "value": 18,
+        "options": {}
+    },
+    "ais.queryRules": {
+        "value": 19,
+        "options": {}
+    },
+    "ais.queryRuleCustomData": {
+        "value": 20,
+        "options": {}
+    },
+    "ais.queryRuleContext": {
+        "value": 21,
+        "options": {}
+    },
+    "ais.range": {
+        "value": 22,
+        "options": {}
+    },
+    "ais.rangeInput": {
+        "value": 23,
+        "options": {}
+    },
+    "ais.rangeSlider": {
+        "value": 24,
+        "options": {}
+    },
+    "ais.ratingMenu": {
+        "value": 25,
+        "options": {}
+    },
+    "ais.refinementList": {
+        "value": 26,
+        "options": {}
+    },
+    "ais.searchBox": {
+        "value": 27,
+        "options": {}
+    },
+    "ais.sortBy": {
+        "value": 28,
+        "options": {}
+    },
+    "ais.stats": {
+        "value": 29,
+        "options": {}
+    },
+    "ais.toggleRefinement": {
+        "value": 30,
+        "options": {}
+    },
+    "ais.voiceSearch": {
+        "value": 31,
+        "options": {}
+    }
+};
 
-var WidgetParams = (exports.WidgetParams = {
-  indexName: 1000,
-  searchClient: 1001,
-  numberLocale: 1002,
-  searchFunction: 1003,
-  initialUiState: 1004,
-  onStateChange: 1005,
-  stalledSearchDelay: 1006,
-  routing: 1007,
-  insightsClient: 1008,
-  container: 1009,
-  placeholder: 1010,
-  autofocus: 1011,
-  searchAsYouType: 1012,
-  showReset: 1013,
-  showSubmit: 1014,
-  showLoadingIndicator: 1015,
-  queryHook: 1016,
-  templates: 1017,
-  cssClasses: 1018,
-  escapeHTML: 1019,
-  transformItems: 1020,
-  attributes: 1021,
-  showMore: 1022,
-  attribute: 1023,
-  escapeFacetValues: 1024,
-  searchParameters: 1025,
-  on: 1026,
-  items: 1027,
-  padding: 1028,
-  precision: 1029,
-  max: 1030,
-  searchable: 1031,
-  searchablePlaceholder: 1032,
-  searchableShowReset: 1033,
-  attributesToSnippet: 1034,
-  snippetEllipsisText: 1035,
-  removeWordsIfNoResults: 1036,
-  scrollTo: 1037,
-  showFirst: 1038,
-  showLast: 1039,
-  pips: 1040,
-  tooltips: 1041,
-});
+var WidgetParams = exports.WidgetParams = {
```

Comment:
would there be a way to make this esm instead of commonjs?

Reply:
Sadly `pbf` only output a commonjs module (issue: https://github.com/mapbox/pbf/issues/97)

Reply:
I think if you change these lines it should work; since they have an issue, they will probably accept the PR

https://github.com/mapbox/pbf/blob/master/compile.js#L94-L97